src/mpi/request/mpir_request.c

```diff
@@ -35,6 +35,24 @@ int MPIR_Request_completion_processing(MPIR_Request * request_ptr, MPI_Status *
                 MPII_SENDQ_FORGET(request_ptr);
                 break;
             }
+        case MPIR_REQUEST_KIND__ANYSRC_RECV:
```

Comment:
I remember I looked around and seems the only real use of `kind` in a request is to be used here for MPIR-layer completion processing (of course I need double check that). So while you are logically correct, practically it just need to be enum that determines proper completion handling.

Reply:
I think that's mostly what I talked myself into in the stream of consciousness above, but I do still want to see the parts that get re-used above moved out into a function. Specifically the code for a regular receive since it's also used in the anysource receive.